WordPress/src/main/java/org/wordpress/android/ui/notifications/NotificationsDetailActivity.java

```diff
@@ -97,22 +94,22 @@ public class NotificationsDetailActivity extends LocaleAwareActivity implements
     @Inject LikesEnhancementsFeatureConfig mLikesEnhancementsFeatureConfig;
     @Inject ListScenarioUtils mListScenarioUtils;
 
-    private String mNoteId;
+    @Nullable private String mNoteId;
     private boolean mIsTappedOnNotification;
 
-    private WPViewPager mViewPager;
-    private ViewPager.OnPageChangeListener mOnPageChangeListener;
-    private NotificationDetailFragmentAdapter mAdapter;
-    private AppBarLayout mAppBarLayout;
-    private Toolbar mToolbar;
+    @Nullable private ViewPager.OnPageChangeListener mOnPageChangeListener;
+    @Nullable private NotificationDetailFragmentAdapter mAdapter;
+
+    @Nullable private NotificationsDetailActivityBinding mBinding = null;
```

Comment:
I'm not suggesting a change here, but just commenting to say that this feels like a good candidate for conversion to Kotlin (though out of scope here 😅). It's unfortunate that we don't have some equivalent of `lateinit` that we could use to avoid all the null guards 🤷‍♂️ . Save that for another day. 🙂

Reply:
Yea, I know @mattmiklic , I had the same feeling while doing this update, but then again I am not best friends with `lateinit` as people have long been starting to avoid using it whenever possible. Then again, on this specific `binding` case, and as per Google's [guidance on activities](https://developer.android.com/topic/libraries/view-binding#activities), using `lateinit` is safe.

FYI: The same does not apply to Google's [guidance on fragments](https://developer.android.com/topic/libraries/view-binding#fragments), since a nullable `binding` is being used there instead. Thus, using a nullable binding for Java activities, until at least we convert that to Kotlin, still sits okay with me, at least for now... 🤷 

PS: About converting to Kotlin, I wouldn't go that far, not only because it might be considered out of scope, but also, because although it might seems easy, a class with almost 700 lines of code deserves a Kotlinization PR of itself.

Reply:
We can do our _own_ thing that is even _better_ than the guidance code. For example, I would replace their usage of the not-null assertion operator (`!!`) with a `checkNotNull`:

```kotlin
private var _binding: ResultProfileBinding? = null
// This property is only valid between onCreateView and
// onDestroyView.
private val binding get() = _binding!!
```

becomes:

```kotlin
private var _binding: ResultProfileBinding? = null
private val binding get() = checkNotNull(_binding) {
  "This property is only valid between onCreateView and onDestroyView."
}
```
 
And this would result in a more appropriate exception (`IllegalStateException`) and trace. We can do the same in Java, of course, but in Kotlin, this is idiomatic and elegant, whereas, in Java, it requires more effort (both to read and write such code).

Reply:
👋 @mkevins !

> TODO

We actually already decided to go another route, to only use the nullable `binding` and not go the `private val binding get() = _binding!!` road.

For more usages of this pattern, search for: `private var binding\: .*Binding\? \= null`

I think that so far this works well and as excepted, without us having much problem with using the `?` operator when needed to access that nullable `binding`, that is, since we try to use extension functions as much as possible in the meanwhile. For example, see `PreviewImageFragment.kt` and the fact that `binding?` is only used during `onDestroyView() { ... }`.

Having said that, comparing `private val binding get() = _binding!!` and the `private val binding get() = checkNotNull(_binding) { ... }`, I indeed like your approach better. And we can definitely revisit if we want to use a nullable `binding`, or a non null going forward, that is, if we find enough value in doing so.

Still, it seems to me all that is (kind of) out of scope for this Java nullability project we are currently undergoing and might only come handy when we start with the Kotlinization project, if and when, right? 🤔